app/controllers/uploads_controller.rb

```diff
@@ -31,9 +31,17 @@ def document_params
     params.require(:documents).map do |document|
       {
         io: document,
-        filename: document.original_filename,
+        filename: derived_filename(document), # You can customize this as needed
```

Comment:
I guess we can't have final name here because topic is unknown, right?
If yes, then why do we need this piece?